Inform about OIDC + LDAP (#448)
Enalean/tuleap-documentation-en: languages/en/administration-guide/users-management/security/user-authentication/openidconnect.rst

The OpenID Connect client plugin allows users to authenticate with an OpenID Connect provider.

.. attention::

    OpenID Connect is meant to delegate authentication to a 3rd party provider with a global goal to no longer have
    credentials management by Tuleap (either in Local DB or LDAP).

>     As soon users identify themselves with an OpenID Connect provider, they will no longer be able to use their existing

Reviewer: ```suggestion
    As soon as users identify themselves with an OpenID Connect provider, they will no longer be able to use their existing
```

>     credentials (if they had some) for:

Reviewer: ```suggestion
    credentials (if they had any) for:
```

.. warning::

    Enabling both LDAP and OpenID Connect is supposed to be temporary for the authentication part. As stated in introduction
    for security and mental health reasons, there should be a single source of truth of user management.

In order to activate LDAP and OpenID Connect in parallel you must define a common field between the two authentication
sources. It means that your OpenID Connect provider should expose on it's ``userinfo`` route a field that match the one
configured as ``$sys_ldap_eduid`` in the LDAP configuration (will be user's ``ldap_id``).

For instance, if your LDAP provider is Microsoft Active Directory, the OpenID Connect provider must expose the ``sAMAccountName``
in ``userinfo``, for instance as ``preferred_name`` or any other field.

When you have this information, you must configure it with ``tuleap`` CLI as system administrator:

.. sourcecode:: shell

    tuleap config-set openidconnectclient_ldap_attribute preferred_name

After having run this command, you can configure your OpenID Connect provider as described in previous sections.

> Please note that Tuleap trust 100% the OpenID Connect provider here, the value set for ``ldap_id`` is taken without any

Reviewer: suggestion: trust 100% => trusts completely
```suggestion
Please note that Tuleap trusts completely the OpenID Connect provider here, the value set for ``ldap_id`` is taken without any
```

form of verification.

Troubleshooting
---------------

You can monitor what's happening by looking at :ref:`LDAP and OpenID Connect logs <platform_logs>`:

> OpenID Connect will complain if is not found in userinfo route:

Reviewer: ```suggestion
OpenID Connect will complain if ``preferred_name`` is not found in the userinfo route:
```
